Feature/zenko 4641

williamlardier opened the pull request:
WIP

Please review https://scality.atlassian.net/wiki/spaces/OS/pages/2892824661/Artesca+XDM+with+multiple+MongoDB+shards+phase+1 before this PR.

Commit:
ZENKO-4641: render more mongodb shard manifests

Commit:
ZENKO-4641: add shard selector to pvc templates

Seragonia committed:
Add the patch for configsvr pvc selector

Issue: ZENKO-4641

Seragonia committed:
Add 6 node 2 shards mode

Issue: ZENKO-4641

bert-e commented:
# Hello williamlardier,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4641 contains:


* `2.8.47`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.12`


Please check the `Fix Version/s` of ZENKO-4641, or the target
branch of this pull request.

Seragonia committed:
Support multiple shards in CI

Issue: ZENKO-4641

Seragonia committed:
Have one podmonitor per shard

Issue: ZENKO-4641

Seragonia committed:
Switch to hard anti affinity preset for mongos

This will help not having two mongos servers on the same node.

Issue: ZENKO-4641

Seragonia committed:
Force P-S-S topology

We do not want to have N shard server for N nodes, because
it would divide the total available resources per shard,
reducing the available memory: we should reduce the need to
rely on the disks as much as possible to improve the performances,
as per official recommendations.

Issue: ZENKO-4641

ghost commented:
Closing: done in https://github.com/scality/Zenko/pull/2219

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4641 contains:


* `2.12.0`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.14`

* `2.11.7`

* `2.12.0`


Please check the `Fix Version/s` of ZENKO-4641, or the target
branch of this pull request.
